File: 1189/solution.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public int maxNumberOfBalloons(String text) {
        Map<Character, Integer> counter = new HashMap<>();
        for(int i = 0; i < text.length(); i++) {
            counter.put(text.charAt(i), counter.getOrDefault(text.charAt(i), 0) + 1);
        }

        int res = Integer.MAX_VALUE;
        int[] counts = {
                counter.getOrDefault('b', 0),
                counter.getOrDefault('l', 0) / 2,
                counter.getOrDefault('o', 0) / 2,
                counter.getOrDefault('n',0)
        };
        for(int count: counts) {
            res = Math.min(count, res);
        }
        return res;
    }
}

File: 1189/solution.ts
function maxNumberOfBalloons(text: string): number {
  const letterCounts: Map<string, number> = new Map();
  for (let letter of text) {
    letterCounts.set(letter, (letterCounts.get(letter) || 0) + 1);
  }
  let res = Infinity;
  for (let letter of "ban") {
    res = Math.min(letterCounts.get(letter) || 0, res);
  }
  res = Math.min(Math.floor((letterCounts.get("o") || 0) / 2), res);
  res = Math.min(Math.floor((letterCounts.get("l") || 0) / 2), res);
  return res;
}
